### PR TITLE
180 add api documentation to GitHub pages

### LINKED ISSUE #180: Add API documentation to github pages
The api documentation is currently built and served by the API.
It would be good to also have this publicly accessible, alongside the code documentation on Github Pages

### PR DESCRIPTION
**Pull request by LukeParky**
Instead of serving documentation for API specs from the backend, provides a github pages website that is always available.

A demonstation of this is shown at https://lukeparky.github.io/Digital-Twins/api.
It is also linked to from the sphinx docs. Demonstation at https://lukeparky.github.io/Digital-Twins.

I've chosen to leave the existing flask-swagger-ui in place so that developers can access the most up-to-date version without affecting the master version.

Closes #180 

## Developer Checklist
- [x] Make code change
- [ ] Update tests
  - [ ] Update / create new tests
  - [x] Ensure these tests have the expected behaviour
  - [x] Test locally and ensure tests are passing
- [x] Update documentation
  - [x] Readme
  - [x] Docstrings
  - [x] Comments
  - [ ] Wiki -- **to be done once merged**

## Reviewer Checklist
- [ ] Check new code for code smells
- [ ] Check new tests
  - [ ] Ensure adequate coverage
  - [ ] Check for code smells within tests
- [ ] Check if documentation needs updating
  - [ ] Readme
  - [ ] Docstrings
  - [ ] Comments
  - [ ] Wiki

### CODE REVIEW

File: autodoc/swagger/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+It includes CSS and OAuth configuration.
+
+`_sample_package.json` is a placeholder sample. You should rename this file, per `Usage` section below, and you should also verify and update this sample's `@latest` compared to the `swagger-ui@latest`
+
+
+#### Usage
+    rename `_sample_package.json` to `package.json`
+    npm install
+    npm start
```
Comment (LukeParky) [inline autodoc/swagger/README.md:14]:
```suggestion
### Automated API documentation website building.

This small webpack project creates the webpages for our API documentation.
It is linked to GitHub Actions to merge with out Python docs.

The source for this documentation is the API spec defined at `src/static/api_documentation.yml`

#### Usage
    npm ci
    npm run build
    
The built documentation can be viewed by opening `dist/index.html` in browser.
```

File: autodoc/swagger/index.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import SwaggerUI from 'swagger-ui'
+import 'swagger-ui/dist/swagger-ui.css';
+
```
Comment (LukeParky) [inline autodoc/swagger/index.js:3]:
```suggestion

// Load API specification into webpage UI
```

File: docs/_sources/index.rst.txt [hunk 3]
```diff
@@ -17,3 +17,10 @@ Indices and tables
 * :ref:`genindex`
 * :ref:`modindex`
 * :ref:`search`
+
+
+Link to my non-Sphinx HTML file
+===============================
+
+To get to my standalone, non-generated HTML file,
+just `click here </./swagger/index.html>`_ now!
```
Comment (LukeParky) [inline docs/_sources/index.rst.txt:26]: Remove all docs commits

**Comment by LukeParky** [comment]
@caseyli92, I have put @Martin20494 as a co-reviewer on this so that he can begin getting accustomed to pull reviews. I won't merge until both of you agree but you will be the main reviewer.

**Review by Martin20494 — COMMENTED** [review]
The changes look really great to me.

**Review by caseyli92 — APPROVED** [review]
Looks good. Remember to remove `Try it out`.

File: .github/workflows/build-documentation.yml [hunk 4]
```diff
@@ -2,34 +2,44 @@ name: Build & Publish Documentation
 on:
   push:
     branches:
-    - master
-    - autodoc
+      - master
+      - autodoc
```
[the remark this answers is not in the document]
Comment (LukeParky) [reply .github/workflows/build-documentation.yml:6]: Resolved this instead by auto indenting everything else

**Review by caseyli92 — APPROVED** [review]
Looks good :)

**Pull request merged**